src/playwright-req-handler.ts

```diff
@@ -59,11 +59,14 @@ function isValidContentType(contentType: string | undefined) {
 /**
  * Generic handler for processing the page content (adapted from: Website Content Crawler).
  */
-export async function requestHandlerPlaywright(
-    context: PlaywrightCrawlingContext<UserData>,
-    settings: PlaywrightScraperSettings,
-) {
+export async function requestHandlerPlaywright(context: PlaywrightCrawlingContext<UserData>) {
     const { request, contentType, page, response, closeCookieModals } = context;
+    const { playwrightScraperSettings: settings, responseId } = request.userData;
+
+    if (!settings) {
+        log.error('Missing playwrightScraperSettings in request.userData');
```

Comment:
If you use typed UserData then TS should ensure you pass it in and you can remove this error. Just make sure whenever you addRequest, you also use the UserData type. If you would need to make separate UserData for each Crawler then I would do it in this PR, less chance we will mess something up and need this error